javaslang-benchmark/src/test/java/javaslang/idiom/ForBenchmark.java

```diff
@@ -0,0 +1,90 @@
+package javaslang.idiom;
+
+import javaslang.JmhRunner;
+import javaslang.collection.Array;
+import org.junit.Test;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static javaslang.API.For;
+import static javaslang.JmhRunner.getRandomValues;
+
+public class ForBenchmark {
+    static final Array<Class<?>> CLASSES = Array.of(
+            ForComprehension1Iterator.class,
+            ForComprehension2Iterators.class
+    );
+
+    @Test
+    public void testAsserts() {
+        JmhRunner.runDebugWithAsserts(CLASSES);
+    }
+
+    public static void main(String... args) {
+        JmhRunner.runNormalNoAsserts(CLASSES);
+    }
+
+
+    @State(Scope.Benchmark)
+    public static class Base {
+        @Param({ "10", "100", "100" })
+        public int CONTAINER_SIZE;
+
+        Integer[] ELEMENTS;
+        List<Integer> ELEMENTS_ARRAY;
+
+        @Setup
+        public void setup() {
+            ELEMENTS = getRandomValues(CONTAINER_SIZE, 0);
+            ELEMENTS_ARRAY = Arrays.asList(ELEMENTS);
+        }
+
+        protected Object consume(Blackhole bh, Integer i) {
+            bh.consume(i.intValue());
+            return null;
+        }
+
+        protected Object consume(Blackhole bh, Integer i, Integer j) {
+            bh.consume(i.intValue());
+            bh.consume(j.intValue());
+            return null;
+        }
+    }
+
+    public static class ForComprehension1Iterator extends Base {
+
+        @Benchmark
+        public void javaslang_for(Blackhole bh) {
+            For(ELEMENTS_ARRAY).yield(i -> consume(bh, i));
+        }
+
+        @Benchmark
+        public void java_for(Blackhole bh) {
+            for (Integer i : ELEMENTS_ARRAY) {
+                consume(bh, i);
+            }
+        }
+    }
+
+    public static class ForComprehension2Iterators extends Base {
```

Comment:
Problem is that the `For` comprehension mostly makes sense for nested iterations, otherwise you could use a regular `map`. I'd like the benchmark to be similar to real-world usage, since it is comparing common idioms IMHO.